### PR TITLE
Irace test

### PR DESCRIPTION
**Pull request by smohsinali**

### CODE REVIEW

File: ConfigSpace/io/irace.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+        conditions = condition.components
+
+    else:
+        logic = None
+
+    for clause in conditions:
+        child = clause.child.name
+        # Findout type of parent
+        pType = "i"
+        if clause.parent.__class__.__name__ == 'CategoricalHyperparameter':
```
Comment (mfeurer) [inline ConfigSpace/io/irace.py]: Is there a reason not to use `isinstance` here?

File: ConfigSpace/io/irace.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+            pType = 'c'
+        elif clause.parent.__class__.__name__ == 'UniformFloatHyperparameter':
+            pType = 'r'
+        elif clause.parent.__class__.__name__ == 'OrdinalHyperparameter':
+            pType = 'o'
+
+        if pType == "i" or pType == "r":
+            if clause.parent.log:
+                if hasattr(clause, 'values'):
+                    clause.values = np.log(clause.values)
```
Comment (mfeurer) [inline ConfigSpace/io/irace.py]: What is the intuition here? Can you add some comments?

File: ConfigSpace/io/irace.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    notequal_template = "%s != %s"
+    equal_template = "%s==%s"
+
+    full_condition = ''
+    child = ''
+    conditions = [condition] # if not a conjunction this makes condition iteratable
+
+    # Check if IRACE can handle the condition
+    if isinstance(condition, OrConjunction):
+        # raise NotImplementedError("IRACE cannot handle OR conditions: %s" %
```
Comment (mfeurer) [inline ConfigSpace/io/irace.py]: Could you please remove all this comments which are no longer valid?

File: ConfigSpace/io/irace.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+        # create a ForbiddenAnd and add all ForbiddenEquals
+        if len(in_statements) > 0:
+            for i, p in enumerate(product(*in_statements)):
+                all_forbidden_clauses = list(p) + other_statements
+                f = ForbiddenAndConjunction(*all_forbidden_clauses)
+                forbidden_lines.append(build_forbidden(f))
+        else:
+            forbidden_lines.append(build_forbidden(forbidden_clause))
+
+    # Add condtions: first conver param_lines to array then search first part of condition in that array
```
Comment (mfeurer) [inline ConfigSpace/io/irace.py]: There is a t missing in `convert`

File: ConfigSpace/io/irace.py [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    if len(forbidden_lines) > 0:
+        for forbidden in forbidden_lines:
+            forbidden_lines_write.write(forbidden + '\n')
+
+    output_fh = open('forbidden.txt', 'w')
+    output_fh.write(forbidden_lines_write.getvalue())
+    output_fh.close()
+
+
+    # Check if the default configuration is a valid configuration!
```
Comment (mfeurer) [inline ConfigSpace/io/irace.py]: Could you actually do what this comment is hinting at?

File: ConfigSpace/io/irace.py (hunk 3 above)
Comment (smohsinali) [inline ConfigSpace/io/irace.py]: removed

File: ConfigSpace/io/irace.py (hunk 1 above)
Comment (smohsinali) [inline ConfigSpace/io/irace.py]: not really, changed to isinstance

File: ConfigSpace/io/irace.py (hunk 2 above)
Comment (smohsinali) [inline ConfigSpace/io/irace.py]: comments added

File: ConfigSpace/io/irace.py (hunk 4 above)
Comment (smohsinali) [inline ConfigSpace/io/irace.py]: fixed the typo

File: ConfigSpace/io/irace.py (hunk 5 above)
Comment (smohsinali) [inline ConfigSpace/io/irace.py]: isnt this checked in class ConfigurationSpace itself

Comment (mfeurer) [inline ConfigSpace/io/irace.py]: Correct!

**Pull request merged**